pandas/tests/arrays/categorical/test_missing.py

```diff
@@ -148,3 +148,22 @@ def test_use_inf_as_na_outside_context(self, values, expected):
             result = pd.isna(DataFrame(cat))
             expected = DataFrame(expected)
             tm.assert_frame_equal(result, expected)
+
+    def test_compare_categorical_with_missing(self):
```

Comment:
can you fully replicate the tests in the OP, e.g. add for == as well.

ideally parameterize this over these 4 cases (or 2 and 2 in-line if easier).

Reply:
Thank you for review. Done.